pkg/resources/provisioning.cattle.io/v1/cluster/validator_test.go

```diff
@@ -200,3 +202,337 @@ func TestValidateMachinePoolName(t *testing.T) {
 		})
 	}
 }
+
+func TestValidateAgentEnvVars(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name          string
+		request       *admission.Request
+		cluster       *v1.Cluster
+		oldCluster    *v1.Cluster
+		shouldSucceed bool
+	}{
+		{
```

Comment:
One other test case that would be good to add: there is an RKEConfig with some agent env vars, but not the AGENT_VAR_DIR.